src/ckan_to_bigquery.py

```diff
@@ -70,34 +81,84 @@ def search_sql(self, sql):
         '''
         NB: table must be full table id ...
         '''
-        # limit the number of results to ckan.datastore.search.rows_max
+        # limit the number of results to ckan.datastore.search.rows_max + 1
+        # (the +1 is so that we know if the results went over the limit or not)
         try:
              # check rows_max parameter set in CKAN config,
              # while testing as microlibrary (not as ckan ext) ckan is unknown
-            from ckan.common import config
             rows_max = int(config.get('ckan.datastore.search.rows_max', 32000))
         except:
-            rows_max = 32000 # set default rows limit 
-        
+            rows_max = 32000 # set default rows limit  
+        sql_initial = sql
         # limit the number of results to return by rows_max
-        sql = 'SELECT * FROM ({0}) AS blah LIMIT {1} ;'.format(sql, rows_max)
-        query = sql
-        query_job = self.bqclient.query(query)
-        rows = query_job.result()
+        sql = 'SELECT * FROM ({0}) AS blah LIMIT {1} ;'.format(sql, rows_max+1)
```

Comment:
I'm not quite sure what this is doing. could you clarify and maybe add a comment in the code so clear to others in future.

Reply:
I'm just trying to be as close as possible with CKAN core structure.
https://github.com/ckan/ckan/blob/master/ckanext/datastore/backend/postgres.py#L1564